cmd/abapEnvironmentCreateTag.go

```diff
@@ -0,0 +1,267 @@
+		if err != nil {
+			errorOccurred = true
+		}
+	}
+	if errorOccurred {
+		message := "At least one tag has not been created"
+		err = errors.New(message)
+	}
+	return err
+}
+
+func createSingleTag(item CreateTagBacklog, index int, telemetryData *telemetry.CustomData, con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+
+	requestBodyStruct := CreateTagBody{RepositoryName: item.repositoryName, CommitID: item.commitID, Tag: item.tags[index].tagName, Description: item.tags[index].tagDescription}
+	requestBodyJson, err := json.Marshal(&requestBodyStruct)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Request body: %s", requestBodyJson)
+	resp, err := abaputils.GetHTTPResponse("POST", con, requestBodyJson, client)
+	if err != nil {
+		errorMessage := "Could not create tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID
+		err = abaputils.HandleHTTPError(resp, err, errorMessage, con)
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Parse response
+	var createTagResponse CreateTagResponse
+	var abapResp map[string]*json.RawMessage
+	bodyText, _ := ioutil.ReadAll(resp.Body)
+
+	if err = json.Unmarshal(bodyText, &abapResp); err != nil {
+		return err
+	}
+	if err = json.Unmarshal(*abapResp["d"], &createTagResponse); err != nil {
+		return err
+	}
+
+	con.URL = con.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull(guid'" + createTagResponse.UUID + "')"
+	err = checkStatus(con, client, com)
+
+	if err == nil {
+		log.Entry().Info("Created tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID)
+	} else {
+		log.Entry().Error("NOT created: Tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID)
+	}
+
+	return err
+}
+
+func checkStatus(con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+	status := "R"
+	pollIntervall := com.GetPollIntervall()
+	count := 0
+	for {
+		count += 1
+		entity, _, err := abaputils.GetStatus("Could not create Tag", con, client)
+		if err != nil {
+			return err
+		}
+		status = entity.Status
+		if status != "R" {
+			if status == "E" {
+				err = errors.New("Could not create Tag")
+			}
+			return err
+		}
+		if count >= 200 {
+			return errors.New("Could not create Tag (Timeout)")
+		}
+		time.Sleep(pollIntervall)
+	}
+}
+
+func prepareBacklog(config *abapEnvironmentCreateTagOptions) (backlog []CreateTagBacklog, err error) {
+
+	if config.Repositories != "" && config.RepositoryName != "" {
+		return nil, errors.New("Configuring the parameter repositories and the parameter repositoryName at the same time is not allowed")
+	}
+
+	descriptor := abaputils.AddonDescriptor{}
+	repos := []abaputils.Repository{}
+	if config.Repositories != "" {
```

Comment:
```suggestion
	if config.Repositories != "" {
		descriptor := abaputils.AddonDescriptor{}
		repos := []abaputils.Repository{}	
```
variables are only used inside the if so I would move the declaration there